pkg/plan/common.go

```diff
@@ -312,18 +327,45 @@ func CreateOrUpdate(
 				log.Info("Resource is ignored", "namespace", rs.Namespace, "name", rs.Name, "kind", kind)
 				return nil
 			}
+			// Only update replicas if we are changing to/from zero, which means the replicaset is being retired/deployed
 			var zero int32 = 0
 			if rs.Spec.Replicas == nil {
 				rs.Spec.Replicas = &zero
 			}
 			if 0 == *typed.Spec.Replicas || 0 == *rs.Spec.Replicas {
 				*rs.Spec.Replicas = *typed.Spec.Replicas
+			} else {
+				// if the only change is replicas, and neither is zero, abort update. Replicas can change so fast in
```

Comment:
Will this work with WPAs that can scale ReplicaSet to 0?

Reply:
I don't think this PR makes a difference in that regard. Existing picchu will keep overwriting `replicas: 0` with default replica count if the revision is deployed as is.